packages/video_player/video_player_web/lib/src/video_player.dart

```diff
@@ -205,8 +205,14 @@ class VideoPlayer {
   /// Moves the playback head to a new `position`.
   ///
   /// `position` cannot be negative.
+  ///
+  /// Does nothing if the new position is the same as the current position.
   void seekTo(Duration position) {
     assert(!position.isNegative);
+    // Don't seek if video is already at position.
+    if (position.inMilliseconds == (_videoElement.currentTime * 1000).round()) {
```

Comment:
The common code with `getPosition()` should be extracted to a shared private helper so that they are guaranteed to stay in sync, since what's important here is that it matches what the higher-level code got from `getPosition`.